tardis/__init__.py

```diff
@@ -33,6 +33,126 @@
 logger.addHandler(console_handler)
 logging.getLogger("py.warnings").addHandler(console_handler)
 
+LOGGING_LEVELS = {
+    "NOTSET": logging.NOTSET,
+    "DEBUG": logging.DEBUG,
+    "INFO": logging.INFO,
+    "WARNING": logging.WARNING,
+    "ERROR": logging.ERROR,
+    "CRITICAL": logging.CRITICAL,
+}
+
+
+class FilterLog(object):
+    """
+    Filter Log Class for Filtering Logging Output
+    to a particular level
+
+    Parameters
+    ----------
+    log_level : logging object
+        allows to have a filter for the
+        particular log_level
+    """
+
+    def __init__(self, log_level):
+        self.log_level = log_level
+
+    def filter(self, log_record):
+        """
+        filter() allows to set the logging level for
+        all the record that are being parsed & hence remove those
+        which are not of the particular level
+
+        Parameters
+        ----------
+        log_record : logging.record
+            which the paricular record upon which the
+            filter will be applied
+
+        Returns
+        -------
+        boolean : True, if the current log_record has the
+            level that of the specified log_level
+            False, if the current log_record doesn't have the
+            same log_level as the specified one
+        """
+        return log_record.levelno == self.log_level
+
+
+# Setting up a list to store the Logging Filters set by logger.addFilter()
+list_of_filter = []
+
+
+def logging_state(log_state, tardis_config, specific):
+    """
+    Function to set the logging configuration for the simulation output
+    Called from within run_tardis()
+    Configured via functional arguments passed through run_tardis() - log_state & specific
+    Configured via YAML parameters under `debug` section - logging_level & specific_logging
+
+    Parameters
+    ----------
+    log_state: str
+        Allows to input the log level for the simulation
+        Uses Python logging framework to determine the messages that will be output
+    specific: boolean
+        Allows to set specific logging levels. Logs of the `log_state` level would be output.
+    """
+    if "debug" in tardis_config:
+        if specific or tardis_config["debug"]["specific"]:
+            specific = True
+        else:
+            specific = False
+
+        if tardis_config["debug"]["log_state"] or log_state:
```

Comment:
Again, we shouldn't be prioritizing `True`, we should be prioritizing an option.  So we should either only check the `log_state` or have `log_state` be a kwarg set to `None`  and use a similar pattern to what is above.

Reply:
Yes, I have changed the way `log_state` now behaves. It has a default value of `None` now 😄